hack/generator/pkg/astmodel/armconversion/shared.go

```diff
@@ -62,17 +62,20 @@ func GetAzureNameProperty(idFactory astmodel.IdentifierFactory) *astmodel.Proper
 
 func getReceiverObjectType(codeGenerationContext *astmodel.CodeGenerationContext, receiver astmodel.TypeName) *astmodel.ObjectType {
 	// Determine the type we're operating on
-	receiverType, err := codeGenerationContext.GetImportedDefinition(receiver)
+	rt, err := codeGenerationContext.GetImportedDefinition(receiver)
 	if err != nil {
 		panic(err)
 	}
 
-	kubeType, ok := receiverType.Type().(*astmodel.ObjectType)
-	if !ok {
-		panic(fmt.Sprintf("receiver for ArmConversionFunction is not of type ObjectType. TypeName: %v, Type %T", receiver, receiverType.Type()))
+	switch receiverType := rt.Type().(type) {
```

Comment:
I think the entire purpose of the wrapper types as we have them (ArmType, StorageType, One/etc) is to give us a way to identify types while we're performing type translations. Information like that certainly isn't ever going to be required at code generation time (because its purpose is for the pipeline phases)?

I could see the argument that there may be other types that wrap struct but can't be translated back to struct, but those types are then by definition not wrapper types, I think?

Reply:
Put another way:

It doesn't seem worth it to me to pay the price (coupling in the code generation code) now, when we don't have any examples where we actually need that functionality. We could go with something that removed that coupling (pipeline stage projecting all the wrapper types away) and always revisit it later if a type came up that didn't work for.

Reply:
I'm torn because it seems as though the actual code generation step is already doing some heavy lifting, but I agree that I shouldn't be making things worse. 

I've created PR #261 to simplify types near the end of the pipeline; once that's merged I'll rebase this branch and simplify the method above.

Reply:
Refactored.